docs/rfcs/017-asset-modified-cleanup.md

```diff
@@ -0,0 +1,140 @@
+# Asset modified cleanup
+
+## Brief
+
+As part of the delivery channels work, it's possible to modify an asset to either change a delivery channel, or update said delivery channel and leave orphaned assets in S3.  This RFC is to discuss how to remove these assets safely and the various permutations of this removal logic.
+
+## API
+
+Currently, when an asset is modified, a message is published to an SNS topic by the `AssetNotificationSender` with the `before` and `after` of the asset.  This process can then fan out to SQS queue which is then handled by a service that cleans up modified assets in a similar process to how `DELETE` requests are handled.
+
+### Changes
+
+As part of this process there will need to be some changes made to the API, which are as follows:
+
+- AssetModifiedMessage needs to be raised whenever an asset is changed. It's currently only happening for single image requests - PUT, POST or PATCH `/customers/{c}/spaces/{s}/images/{i}` as well as `/reingest` too.
+  - Needs to happen for bulk operations (batch PATCH + queue).
+  - There's already logic to handle batch sending of notifications, just need to get list of messages to send.
+- It's possible to send `AssetModified` messages that aren't required to be ingested (such as metadata changes) and this is primarily controlled by `processAssetResult.RequiresEngineNotification`. In order to reduce churn, an attribute should be added to the request that indicates the asset will be ingested by engine
+  - This attribute should be called something like `EngineNotified`
+- Asset requires the `DeliveryChannelPolicyId` to work out differences in policies (this should be there already)
+- Asset requires `roles` as changes to roles can mean the `info.json` needs to be removed
+- API should not be responsible for deciding when cleanup is conducted
+
+## AWS
+
+As part of the changes, there will need to be some changes to the AWS estate to support the changes to asset modified.  Primarily, this will be updating the current SNS topic and adding an SQS listener to listen for asset modified notifications.  The queue should forward all requests to the listener as there's a possibility 
```

Comment:
Last sentance isn't finished `"The queue should forward all requests to the listener as there's a possibility..."`